uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/speed/SpeedProgressView.kt

```diff
@@ -0,0 +1,216 @@
+        set(value) {
+            field = value
+            foregroundPaint.strokeWidth = value.toFloat()
+            backgroundPaint.strokeWidth = value.toFloat()
+            invalidate()
+        }
+
+    var segmentSpacing = resources.getFraction(R.fraction.speedProgressViewSegmentSpacing, 1, 1)
+        set(value) {
+            field = value
+            invalidate()
+        }
+
+    init {
+        attrs?.let { attributeSet ->
+            @Suppress("Recycle")
+            context.obtainStyledAttributes(
+                attributeSet,
+                R.styleable.SpeedProgressView,
+                defStyleAttr,
+                defStyleRes
+            ).also {
+                roundSegmentEdges = it.getBoolean(
+                    R.styleable.SpeedProgressView_roundSegmentEdges, roundSegmentEdges
+                )
+                segmentBackgroundColor = it.getInt(
+                    R.styleable.SpeedProgressView_segmentBackgroundColor, segmentBackgroundColor
+                )
+                segmentForegroundColors = resources.getIntArray(
+                    it.getResourceId(
+                        R.styleable.SpeedProgressView_segmentForegroundColors, R.array.speedProgressForegroundColors
+                    )
+                )
+                segmentAngle = it.getInt(
+                    R.styleable.SpeedProgressView_segmentAngle, segmentAngle.toInt()
+                ).toFloat()
+                segmentThickness = it.getDimensionPixelSize(
+                    R.styleable.SpeedProgressView_segmentThickness, segmentThickness
+                )
+                segmentSpacing = it.getFraction(
+                    R.styleable.SpeedProgressView_segmentSpacing, 1, 1, segmentSpacing
+                )
+            }.recycle()
+        }
+    }
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+
+        val min = min(
+            getDefaultSize(suggestedMinimumWidth, widthMeasureSpec),
+            getDefaultSize(suggestedMinimumHeight, heightMeasureSpec)
+        )
+        setMeasuredDimension(min, min)
+
+        segmentForegroundGradient = createForegroundGradient(segmentForegroundColors)
+
+        val left = segmentThickness.toFloat() / 2
+        val top = segmentThickness.toFloat() / 2
+        val right = min - segmentThickness.toFloat() / 2
+        val bottom = min - segmentThickness.toFloat() / 2
+        oval.set(left, top, right, bottom)
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        canvas.rotate(angleShift, width.toFloat() / 2, height.toFloat() / 2)
+
+        val stepAngle = segmentAngle * segmentSpacing
+        val stepWithSpacing = segmentAngle + stepAngle
+        val rest = angleMax.rem(stepWithSpacing)
+        val startEndGap = rest / 2
+        val drawStartPoint = angleStartPoint + startEndGap
+        val drawTolerance = 0.1f
```

Comment:
dali by sa niektore tie premenne inlinovat, niektore vytiahnut von (napr tato `drawTolerance` nie je konstanta nahodou?), aby sa nerobilo vsetko v `onDraw`, aj ked to zas az tak asi neovplyvni

Reply:
tak dala by se reálně vytáhnout jen ta `drawTolerance`, ale chtěl jsem to mít už takto po kupě, nevím zda to stojí za to kvůli té jedné to roztrhávat. inlinovat by se dal zase jen rest, stojí to za to? takto je jasné co přesně co k čemu slouží.

Reply:
tak ja hlavne kvoli tomu, ze je to v `onDraw`, ved vies ako... okrem `drawTolerance`, ktora je const, by sa dalo inlineovat `drawEndPoint` na 172 (pouziva sa to len na 1 mieste - akurat by sa riadok 173 predlzil kozmeticky no...)

ale chapem, ze ches mat po kope. ale tu `drawTolerance` by som isto dal ako konstantu. to mi ani nejak tam nesedi, ze by to muselo byt pokope